yolov3_tf2/dataset.py

```diff
@@ -175,21 +214,100 @@ def build_example(xml_data_dict, bytes_image, key):
         'image/object/difficult': tf.train.Feature(int64_list=tf.train.Int64List(value=xml_data_dict['difficult']))
     }))
 
+def open_image(annotation, images_dir, xml_data_dict):
 
-def parse_xml(xml):
-    if not len(xml):
-        return {xml.tag: xml.text}
-    result = {}
-    for child in xml:
-        child_result = parse_xml(child)
-        if child.tag != 'object':
-            result[child.tag] = child_result[child.tag]
-        else:
-            if child.tag not in result:
-                result[child.tag] = []
-            result[child.tag].append(child_result[child.tag])
-    return {xml.tag: result}
+    img_path = os.path.join(
+        images_dir, annotation['filename'].replace('set_01', '').replace(".xml", ".jpg")
+    )
+
+    raw_image = open(img_path, 'rb').read()
+    key = hashlib.sha256(raw_image).hexdigest()
+
+    if raw_image[0] != 255 and raw_image[0] != 137:
+        logging.warning(f"raw {raw_image[0]}")
+        logging.warning(f"bad image {img_path}")
+        raise Exception("bad image")
+    
+    return raw_image, key
+
+
+def augment_dataset_generator(file_pattern, class_file, size, anchors, anchor_masks):
+
+    augmentation_list = build_default_augmentation_pipeline()
+
+    LINE_NUMBER = -1  # TODO: use tf.lookup.TextFileIndex.LINE_NUMBER
+    class_table = tf.lookup.StaticHashTable(tf.lookup.TextFileInitializer(
+        class_file, tf.string, 0, tf.int64, LINE_NUMBER, delimiter="\n"), -1)
+
+    files = tf.data.Dataset.list_files(file_pattern)
+    dataset = files.flat_map(tf.data.TFRecordDataset)
+
+    for instance in dataset:
+
+        example = tf.io.parse_single_example(instance, IMAGE_FEATURE_MAP)
+
+        raw_image = tf.image.decode_image(example['image/encoded'], channels=3)
+
+        x_train = tf.image.resize(raw_image, (size, size))
+        #print(np.asarray(example['image/object/bbox/xmin']), end='x_obj\n')
+        class_text = tf.sparse.to_dense(example['image/object/class/text'], default_value='')
+        labels = tf.cast(class_table.lookup(class_text), tf.float32)
+
+        y_train = tf.stack([
+            tf.sparse.to_dense(example['image/object/bbox/xmin']),
+            tf.sparse.to_dense(example['image/object/bbox/ymin']),
+            tf.sparse.to_dense(example['image/object/bbox/xmax']),
+            tf.sparse.to_dense(example['image/object/bbox/ymax']),
+            labels
+        ], axis=1)
+
+        y_train = add_pads_to_tensor(y_train, 100)
+
+        yield transform_images(x_train, size), y_train
+
+        for (transformation, _) in augmentation_list:
+            yield apply_transformation(transformation, raw_image, y_train, size, anchors, anchor_masks)
+
+
+def apply_transformation(transformation, raw_image, y_train, size, anchors, anchor_masks):
+
+    img = cv2.cvtColor(raw_image.numpy(), cv2.COLOR_BGR2RGB)
+
+    img = img.astype(np.uint8)
```

Comment:
isso é um pouco preocupante, não fazemos `img /= 255` em algum lugar? se for antes disso vai simplesmente zerar a imagem

Reply:
Fazemos depois, para a aumentação fazemos na linha do yield.

Reply:
Nessa função, no return.

Reply:
f15bd52 removido o a tranformação para int como combinado, já que era desnecessária.

Reply:
bccfacc Assim como comentado, haviamos 2 resizes a mais.